### PR TITLE
Use SendInput instead of WinForms' SendKeys

### LINKED ISSUE #19530: CSharpNavigateTo.NavigateTo is flakey.
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/4219/consoleFull#9716893042d31e50d-1517-49fc-92b3-2ca637122019

      Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpNavigateTo.NavigateTo [FAIL]

VB test has also been disabled.

### LINKED ISSUE #30544: Intermittent integration test failures: BasicNavigationBar.CodeSpit and BasicNavigationBar.VerifyNavBar
See https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/4564/artifact/Binaries/debug/Logs/Microsoft.VisualStudio.LanguageServices.IntegrationTests.dll.out.log

```
Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.CodeSpit [STARTING]
      System.NullReferenceException : Object reference not set to an instance of an object.
      Stack Trace:
        
        Server stack trace: 
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.ReflectionExtensions.ObjectExtensions.GetFieldValue(Object instance, String fieldName)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.ReflectionExtensions.ObjectExtensions.GetFieldValue[FieldType](Object instance, String fieldName)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetNavigationBarComboBoxes(IWpfTextView textView)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.<>c__DisplayClass54_0.<GetSelectedNavBarItem>b__0(IWpfTextView v)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass9_0`1.<ExecuteOnActiveView>b__0()
           at System.Windows.Threading.DispatcherOperation`1.InvokeDelegateCore()
           at System.Windows.Threading.DispatcherOperation.InvokeImpl()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Windows.Threading.DispatcherOperation.Wait(TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.InvokeImpl(DispatcherOperation operation, CancellationToken cancellationToken, TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.Invoke[TResult](Func`1 callback, DispatcherPriority priority, CancellationToken cancellationToken, TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.Invoke[TResult](Func`1 callback, DispatcherPriority priority)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.InvokeOnUIThread[T](Func`1 action)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.ExecuteOnActiveView[T](Func`2 action)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetSelectedNavBarItem(Int32 comboBoxIndex)
           at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
           at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)
        
        Exception rethrown at [0]: 
           at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
           at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetSelectedNavBarItem(Int32 comboBoxIndex)
        /_/src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs(279,0): at Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess.Editor_OutOfProc.GetTypeNavBarSelection()
        /_/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicNavigationBar.cs(104,0): at Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.VerifyLeftSelected(String expected)
        /_/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicNavigationBar.cs(80,0): at Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.CodeSpit()
    Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.CodeSpit [FINISHED] Time: 0.572537s
```

```
    Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.VerifyNavBar [STARTING]
      System.NullReferenceException : Object reference not set to an instance of an object.
      Stack Trace:
        
        Server stack trace: 
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.ReflectionExtensions.ObjectExtensions.GetFieldValue(Object instance, String fieldName)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.ReflectionExtensions.ObjectExtensions.GetFieldValue[FieldType](Object instance, String fieldName)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetNavigationBarComboBoxes(IWpfTextView textView)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.<>c__DisplayClass54_0.<GetSelectedNavBarItem>b__0(IWpfTextView v)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass9_0`1.<ExecuteOnActiveView>b__0()
           at System.Windows.Threading.DispatcherOperation`1.InvokeDelegateCore()
           at System.Windows.Threading.DispatcherOperation.InvokeImpl()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
           at System.Windows.Threading.DispatcherOperation.Wait(TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.InvokeImpl(DispatcherOperation operation, CancellationToken cancellationToken, TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.Invoke[TResult](Func`1 callback, DispatcherPriority priority, CancellationToken cancellationToken, TimeSpan timeout)
           at System.Windows.Threading.Dispatcher.Invoke[TResult](Func`1 callback, DispatcherPriority priority)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.InvokeOnUIThread[T](Func`1 action)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.ExecuteOnActiveView[T](Func`2 action)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetSelectedNavBarItem(Int32 comboBoxIndex)
           at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
           at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)
        
        Exception rethrown at [0]: 
           at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
           at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
           at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.GetSelectedNavBarItem(Int32 comboBoxIndex)
        /_/src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs(279,0): at Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess.Editor_OutOfProc.GetTypeNavBarSelection()
        /_/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicNavigationBar.cs(104,0): at Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.VerifyLeftSelected(String expected)
        /_/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicNavigationBar.cs(40,0): at Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.VerifyNavBar()
    Roslyn.VisualStudio.IntegrationTests.Basic.BasicNavigationBar.VerifyNavBar [FINISHED] Time: 0.4626095s
```

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #19530
Fixes #30544

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs [hunk 1]
```diff
@@ -108,6 +119,8 @@ private void StopServiceCallback(object sender, EventArgs e)
         {
             if (_stopMenuCmd.Enabled)
```
Comment (jaredpar) [inline src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: Can this flag be reset during test execution? If so then I would use a separate flag to track if we added `CurrentDomainAssemblyResolve`

File: src/VisualStudio/IntegrationTest/TestUtilities/Input/AbstractSendKeys.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                        dwFlags = dwFlags,
+                        time = 0,
+                        dwExtraInfo = IntPtr.Zero
+                    }
+                };
+            }
+
+            if (IsExtendedKey(virtualKey))
+            {
+                input.ki.dwFlags |= NativeMethods.KEYEVENTF_EXTENDEDKEY;
```
Comment (jaredpar) [inline src/VisualStudio/IntegrationTest/TestUtilities/Input/AbstractSendKeys.cs]: >KEYEVENTF_EXTENDEDKEY [](start = 50, length = 21)

Is this flag valid when using a virtual key? The docs are ambiguous here but the language suggests it's only valid when applied to a scan code.

[https://docs.microsoft.com/en-us/windows/desktop/api/winuser/ns-winuser-tagkeybdinput](https://docs.microsoft.com/en-us/windows/desktop/api/winuser/ns-winuser-tagkeybdinput)

**Comment by jaredpar** [comment]
Done with review pass (iteration 5)

File: src/VisualStudio/IntegrationTest/TestUtilities/Input/AbstractSendKeys.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestUtilities/Input/AbstractSendKeys.cs]: It's required for scan codes, making the change itself a bug fix for that case. The use of this flag for other cases matches the behavior that was always used in this library so it's not really a change.

File: src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: I don't think so, but I also don't think this command is ever used. We only enable the assembly resolve handler when starting integration tests, so it's outside any user path.

**Pull request merged**